modules/gridmap/grid_map.cpp

```diff
@@ -333,6 +333,24 @@ int GridMap::get_cell_item_orientation(int p_x, int p_y, int p_z) const {
 	return cell_map[key].rot;
 }
 
+Vector3 GridMap::world_to_map(const Vector3 &p_world_pos) const {
+	Vector3 map_pos = p_world_pos / cell_size;
+	map_pos.x = floor(map_pos.x);
+	map_pos.y = floor(map_pos.y);
+	map_pos.z = floor(map_pos.z);
+	return map_pos;
+}
+
+Vector3 GridMap::map_to_world(int p_x, int p_y, int p_z) const {
```

Comment:
I'm not convinced by this signature myself. In other parts of this file cell coordinates are expressed as 3 separate integers. However, in `world_to_map()` we are forced to return cell coordinates as a `Vector3`. It does feel inconsistent but I understand having 3 separate parameters is the way to enforce cell coordinates are integers....

What do people think?

Reply:
I'd say you could use Vector3 to be consistent with the TileMap API.

Then, TileMap also has this mix of `x, y` and `Vector2` methods, with the vector methods with a "v" suffix, like `get_cellv`. Too bad we can't have proper overloading and drop this hack :D
Maybe we should review both APIs before 3.0-beta and make them more consistent, but for the scope of this PR just using Vector3 in map_to_world should be fine.